Rancher PAYG offerings for AWS and Azure (#1008)
rancher/rancher-docs: docs/integrations-in-rancher/cloud-marketplace/aws-marketplace-payg-integration/troubleshooting.md

- Logs from the containers of pods of the job: `kubectl logs -l job-name=<job name> -n <namespaces>`

## Recovery from Failed Pods

If any of the pods aren't running, check the rancher-cloud Pod:

```shell
kubectl get pods --all-namespaces | grep rancher-cloud
```

If the rancher-cloud pod is in an Error state, wait for the pod to be deleted. This should take about one minute.

Fix the problem and run:  

```shell
helm upgrade -n cattle-rancher-csp-deployer-system rancher-cloud --create-namespace \
oci://709825985650.dkr.ecr.us-east-1.amazonaws.com/suse/{{repository}}/rancher-cloud-helm/rancher-cloud --install \
  --version {{chart_version}} \
  --set rancherHostname=$HOST_NAME \
  --set rancherServerURL=https://$HOST_NAME \
  --set rancherReplicas=$REPLICAS \
  --set global.aws.accountNumber=$AWS_ACCOUNT_ID \
  --set global.aws.roleName=$ROLE_NAME
```

## Rancher Usage Record Not found

When you attempt to retrive a usage record, you might see the following message:

```shell
Error from server (NotFound): cspadapterusagerecords.susecloud.net "rancher-usage-record" not found" Check Configuration, Retrieve generated configuration csp-config
```

To resolve the error, run:

```shell

> kubectl get cm -n cattle-csp-billing-adapter-system csp-config -o yaml

Reviewer: ```suggestion
kubectl get configmap -n cattle-csp-billing-adapter-system csp-config -o yaml
```

Other commands use the resource's full name.

---
title: Troubleshooting
---

> ## Troubleshooting

Reviewer 2: ```suggestion
```

This would look like a duplicate of the title when published

This section contains information to help you troubleshoot issues when installing Rancher Prime PAYG.

> Jobs and Pods:

Reviewer 2: ```suggestion
## Jobs and Pods
```

> Check that pods or jobs have status Running/Completed
> 
> 
> ```

Reviewer 2: ```suggestion
Check the status of pods or jobs:

```shell
```

kubectl get pods --all-namespaces
```

> If a pod is not in Running state, you can debug into the root cause by running:

Reviewer 2: ```suggestion
If a pod is not in a Running state, you can attempt to find the root cause with the following commands:
```

Suggest formatting the next few lines as bulleted list

Example --

- Describe pods: `kubectl describe pod <pod-name> -n <namespace>`

Describe pod

```
kubectl describe pod <pod name> -n <namespaces>
```

Pod container logs

```
kubectl logs <pod name> -n <namespaces>
```

Describe job

```
kubectl describe job <job name> -n <namespaces>
```

Logs from the containers of pods of the job

```
kubectl logs -l job-name=<job name> -n <namespaces>
```

> ###Recovery from Failed Pods

Reviewer 2: ```suggestion
## Recovery from Failed Pods
```

> If any of the pods are not Running : 

Reviewer 2: ```suggestion
If any of the pods aren't running, check the rancher-cloud Pod:
```

If rancher-cloud Pod is just an example, suggest using `grep <pod name>` instead of `grep rancher-cloud`

> Check the rancher-cloud Pod

Reviewer 2: ```suggestion
```

```
kubectl get pods --all-namespaces | grep rancher-cloud
```

> If the rancher-cloud pod is in Error state, wait for the pod to be deleted which takes approx 1 min after it goes in Error State.

Reviewer 2: ```suggestion
If the rancher-cloud pod is in an Error state, wait for the pod to be deleted. This should take about one minute.
```

> Fix the problem and execute 

Reviewer 2: ```suggestion
Fix the problem and run:
```

> ```

Reviewer 2: ```suggestion
```shell
```

helm upgrade -n cattle-rancher-csp-deployer-system rancher-cloud --create-namespace \
oci://709825985650.dkr.ecr.us-east-1.amazonaws.com/suse/{{repository}}/rancher-cloud-helm/rancher-cloud --install \
  --version {{chart_version}} \
  --set rancherHostname=$HOST_NAME \
  --set rancherServerURL=https://$HOST_NAME \
  --set rancherReplicas=$REPLICAS \
  --set global.aws.accountNumber=$AWS_ACCOUNT_ID \
  --set global.aws.roleName=$ROLE_NAME
```

> ###Rancher Usage Record Not found:

Reviewer 2: ```suggestion
## Rancher Usage Record Not found:
```

> Error:
> 
> ```

Reviewer 2: ```suggestion
When you attempt to retrive a usage record, you might see the following message:

```shell
```

Error from server (NotFound): cspadapterusagerecords.susecloud.net "rancher-usage-record" not found"
 Check Configuration, Retrieve generated configuration csp-config
```

> Solution:

Reviewer 2: ```suggestion
To resolve the error, run:
```

> ```

Reviewer 2: ```suggestion
```shell
```

kubectl get cm -n cattle-csp-billing-adapter-system csp-config -o yaml
```

> if a configuration is not listed, you can debug into the root cause by checking the pod status and log (Refer Jobs and Pods section).

Reviewer 2: ```suggestion
If a configuration is not listed, you can attempt to find the root cause by checking the pod status and log. See [Jobs and Pods](#jobs-and-pods) for more details.
```
